common/src/services/sync.service.ts

```diff
@@ -286,15 +287,31 @@ export class SyncService implements SyncServiceAbstraction {
 
         await this.cryptoService.setEncKey(response.key);
         await this.cryptoService.setEncPrivateKey(response.privateKey);
-        await this.cryptoService.setOrgKeys(response.organizations);
+        await this.cryptoService.setProviderKeys(response.providers);
+        await this.cryptoService.setOrgKeys(response.organizations, response.providerOrganizations);
         await this.userService.setSecurityStamp(response.securityStamp);
         await this.userService.setEmailVerified(response.emailVerified);
 
         const organizations: { [id: string]: OrganizationData; } = {};
         response.organizations.forEach(o => {
             organizations[o.id] = new OrganizationData(o);
         });
-        return await this.userService.replaceOrganizations(organizations);
+
+        const providers: { [id: string]: ProviderData; } = {};
+        response.providers.forEach(p => {
+            providers[p.id] = new ProviderData(p);
+        });
+
+        response.providerOrganizations.forEach(o => {
+            if (organizations[o.id] == null) {
+                organizations[o.id] = new OrganizationData(o);
+                organizations[o.id].isProviderUser = true;
+            }
+        });
+        return Promise.all([
```

Comment:
Just want to ensure that on the server-side, we're not going to have modification date collisions on the Organization table for any updates, or table locks if these aren't done in sequence (vs. parallel).

Reply:
This only sets the values locally. We don't send any data to the server.